rewrite-core/src/main/java/org/openrewrite/config/Environment.java

```diff
@@ -141,8 +145,30 @@ public Collection<RecipeDescriptor> listRecipeDescriptors() {
         return result;
     }
 
+    @Deprecated
     public Recipe activateRecipes(Iterable<String> activeRecipes) {
-        Map<String, Recipe> recipesByName = listRecipes().stream().collect(toMap(Recipe::getName, identity()));
+        List<String> asList = new ArrayList<>();
+        for (String activeRecipe : activeRecipes) {
+            asList.add(activeRecipe);
+        }
+        return activateRecipes(asList);
+    }
+
+    public Recipe activateRecipes(Collection<String> activeRecipes) {
```

Comment:
Why does Iterable need to be narrowed to Collection?

Reply:
I think I tend the other way -- make parameters `Iterable` unless there is a good reason for it to be a list or collection. Though clearly haven't done this consistently.